pypeit/spectrographs/gemini_gnirs.py

```diff
@@ -435,44 +567,250 @@ def spec_min_max(self):
+        off1 = 0.
+        off2 = 0.
+        off1 /= binspec
+        off2 /= binspat
+        crpix1 += off1
+        crpix2 += off2
+
+        # Create a new WCS object.
+        msgs.info("Generating GNIRS IFU WCS")
+        w = wcs.WCS(naxis=3)
+        w.wcs.equinox = hdr['EQUINOX']
+        w.wcs.name = 'GNIRS IFU'
+        w.wcs.radesys = 'FK5'
+        # Insert the coordinate frame
+        w.wcs.cname = ['RA', 'DEC', 'Wavelength']
+        w.wcs.cunit = [units.degree, units.degree, units.Angstrom]
+        w.wcs.ctype = ["RA---TAN", "DEC--TAN", "WAVE"]  # Note, WAVE is in vacuum
+        w.wcs.crval = [ra, dec, wave0]  # RA, DEC, and wavelength zeropoints
+        w.wcs.crpix = [crpix1, crpix2, crpix3]  # RA, DEC, and wavelength reference pixels
+        w.wcs.cd = np.array([[cd11, cd12, 0.0], [cd21, cd22, 0.0], [0.0, 0.0, dwv]])
+        w.wcs.lonpole = 180.0  # Native longitude of the Celestial pole
+        w.wcs.latpole = 0.0  # Native latitude of the Celestial pole
+
+        return w
+
+    def get_datacube_bins(self, slitlength, minmax, num_wave):
+        r"""
+        Calculate the bin edges to be used when making a datacube.
 
+        Args:
+            slitlength (:obj:`int`):
+                Length of the slit in pixels
+            minmax (`numpy.ndarray`_):
+                An array with the minimum and maximum pixel locations on each
+                slit relative to the reference location (usually the centre
+                of the slit). Shape must be :math:`(N_{\rm slits},2)`, and is
+                typically the array returned by
+                :func:`~pypeit.slittrace.SlitTraceSet.get_radec_image`.
+            num_wave (:obj:`int`):
+                Number of wavelength steps.  Given by::
+                    int(round((wavemax-wavemin)/delta_wave))
+
+        Args:
+            :obj:`tuple`: Three 1D `numpy.ndarray`_ providing the bins to use
+            when constructing a histogram of the spec2d files. The elements
+            are :math:`(x,y,\lambda)`.
+        """
+        # TODO :: The HRIFU might have 25 slits with 13 being the reference
+        xbins = np.arange(1 + 21) - 11.0 - 0.5  # 21 is for 21 slices, and 11 is the reference slit
+        ybins = np.linspace(np.min(minmax[:, 0]), np.max(minmax[:, 1]), 1+slitlength) - 0.5
+        spec_bins = np.arange(1+num_wave) - 0.5
+        return xbins, ybins, spec_bins
+
+    def pypeit_file_keys(self):
+        """
+        Define the list of keys to be output into a standard ``PypeIt`` file.
+
+        Returns:
+            :obj:`list`: The list of keywords in the relevant
+            :class:`~pypeit.metadata.PypeItMetaData` instance to print to the
+            :ref:`pypeit_file`.
+        """
+        pypeit_keys = super().pypeit_file_keys()
+        # TODO :: Think about this... do we even need these extra file keys if skysub is done during the reduction
+        #pypeit_keys += ['calib', 'comb_id', 'bkg_id']
```

Comment:
The keys `comb_id` and `bkg_id` are added when we use the `-b` flag with `pypeit_setup`, we don't add them by default. I believe  `calib` is now always added.
Could you add here the key `filter`? It seems it's a useful information to have in the pypeit file